### PR TITLE
Update Evmos

### PR DESCRIPTION
**Pull request by ljoss17**
Evmos compatible version in Cosmos directory are `v12.1.0`, `v12.1.1` and `v12.1.2`, https://cosmos.directory/evmos/chain.
This PR updates Evmos version from `v9.1.0` to `v12.1.2`

### CODE REVIEW

**Comment by romac** [comment]
Looks like the same issue as https://gitlab.com/thorchain/thornode/-/issues/1250

File: resources.nix [hunk 1]
```diff
@@ -92,15 +92,9 @@
         name = "evmos";
         version = "v12.1.2";
```
Comment (romac) [inline resources.nix:93]: Let's also update the version here. Curious why the version went down, do you know anything about that?

**Comment by JonathanLorimer** [comment]
Just for posterity the issue was that nix uses `go mod -vendor` which [prunes all dependencies that aren't go sources](https://github.com/golang/go/issues/26366) and therefore does not include (statically?) linked c bits that go code may depend on. 
The solution is to use gomod2nix which pre-fetches go dependencies by parsing the `go.mod` file. This adds an additional step to updating packages that require this, but its a fair price to pay to have it all work!

**Review by romac — APPROVED** [review]
🙏 🚀 

@JonathanLorimer Thanks for getting to the bottom of this, it had completely stumped us for months 😅

**Pull request merged**